motor_control/throttle/button_control.ino

```diff
@@ -34,17 +39,38 @@ int ButtonThrottle::mappingFunction(int time_pressed, m_map_type mapType)
       {
         return 0;
       }
-      if (std::exp(ratio) <= 100)
+      if (std::exp(ratio) <= m_maxValue)
       {
-        return std::exp(ratio); //This function has no reliance on m_timeToFull, it simply follows an e exponential curve up til 100. Takes about 4.6 seconds to reach max speed
+        m_speedOfCar = std::exp(ratio);
+        return m_speedOfCar; //This function has no reliance on m_timeToFull, it simply follows an e exponential curve up til maximum value. Takes about 5.5 seconds to reach max speed(assuming 255 is the value we set)
       }
     }
     else
     {
-      return 100;
+      m_speedOfCar = m_maxValue;
+      return m_speedOfCar;
     }
   }
+  }
 }
+int mappingFunctionDeceleration(int time_NotPressed)
+{
+   if(m_isPressed == false)
+  {
+    if(m_speedOfCar ==0)
+    {
+      return m_speedOfCar;
+    }
+     double seconds_conversion = time_notPressed/1000;
+  if(seconds_conversion >= m_delaySpeed) 
```

Comment:
indentation

Reply:
Using VSCode with auto-format on save would really help with this!